### PR TITLE
 Migrate meshery-ui tracking to google tag manager

### PR DESCRIPTION
**Pull request by ayushthe1**
**Notes for Reviewers**

This PR attempts to migrate from `gtag.js` to `google tag manager` for tracking meshery-ui data. Also, this PR shouldn't be merged till it's in a draft state.

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: ui/pages/_document.js [hunk 1]
```diff
@@ -3,34 +3,6 @@ import PropTypes from "prop-types";
...
-  }
-
-  fetch("/api/user/prefs", { credentials: 'include' })
-    .then((res) => res.json())
-    .then((res) => {
-      if (res?.anonymousUsageStats) {
-        gtag("js", new Date());
-        gtag("config", "G-8Q51RLT8TZ", {
-          page_path: window.location.pathname,
-        });
```
Comment (ayushthe1) [inline ui/pages/_document.js:28]: Do we need this logic of firing the tag only when `res?.anonymousUsageStats` is true ?

Comment (leecalcote) [reply ui/pages/_document.js:28]: @Udit-takkar might have some insight.

Comment (Udit-takkar) [reply ui/pages/_document.js:28]: we can directly call (without checking res?.anonymousUsageStats)  gtag  like here https://developers.google.com/tag-platform/gtagjs/install to track all users data anonymously. 

maybe this was added to allow users to opt out of anonymous tracking by checking user preferences and only then track the user behaviour

Comment (ayushthe1) [reply ui/pages/_document.js:28]: Thanks @Udit-takkar for the info :+1: Is there any other change you would suugest ?

Comment (leecalcote) [reply ui/pages/_document.js:28]: We track by default. Users can opt out under their User Preferences.

Is this PR complete or does it need changes @Udit-takkar?

Comment (Udit-takkar) [reply ui/pages/_document.js:28]: @leecalcote @ayushthe1  ah sorry didn't saw this comment before.

Like you said that "you track by default but let user opt out in user preference". i can see this functionality in the removed code but i don't see this in the new added code that we are fetching user preferences and checking anonymousUsageStats and only then  tracking events to Google Analytics to record the page load and configuration information.

File: ui/pages/_document.js [hunk 2]
```diff
@@ -43,34 +15,35 @@ class MesheryDocument extends Document {
           <meta name="referrer" content="no-referrer" />
           <link rel="icon" href="/static/favicon.png" />
 
+          {/* Google Tag Manager */}
+          <script dangerouslySetInnerHTML={{
+            __html: `(function(w,d,s,l,i){w[l]=w[l]||[];w[l].push({'gtm.start': new Date().getTime(),event:'gtm.js'});var f=d.getElementsByTagName(s)[0], j=d.createElement(s),dl=l!='dataLayer'?'&l='+l:'';j.async=true;j.src= 'https://www.googletagmanager.com/gtm.js?id='+i+dl;f.parentNode.insertBefore(j,f);})(window,document,'script','dataLayer','GTM-WR7595H');`
+          }} />
+          {/* End Google Tag Manager */}
```
Comment (Udit-takkar) [inline ui/pages/_document.js:22]: we can do this here if we want users to opt out based on user preferences

```
<script dangerouslySetInnerHTML={{
  __html: `(function(w,d,s,l,i){
    w[l]=w[l]||[];
    w[l].push({'gtm.start': new Date().getTime(),event:'gtm.js'});
    var f=d.getElementsByTagName(s)[0],
        j=d.createElement(s),
        dl=l!='dataLayer'?'&l='+l:'';
    j.async=true;
    j.src='https://www.googletagmanager.com/gtm.js?id='+i+dl;
    f.parentNode.insertBefore(j,f);

    // Fetch user preferences
    fetch("/api/userPreferences", { credentials: 'include' })
      .then((res) => res.json())
      .then((res) => {
        if (res && res.trackData === false) {
          // User opted out of tracking, disable GTM
          w[l] = []; // Clear the dataLayer array
          w['ga-disable-'+i] = true; // Disable Google Analytics tracking
        }
      });
  })(window,document,'script','dataLayer','GTM-WR7595H');`
}} />
```

**Review by Udit-takkar — APPROVED** [review]

**Review by Chadha93 — APPROVED** [review]

**Pull request merged**